src/orders/controller/orders.controller.ts

```diff
@@ -1,13 +1,28 @@
-import { Body, Controller, Delete, Get, Param, Post } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
 import { OrderMapper } from '../mapper/order.mapper';
 import { OrdersService } from '../service/orders.service';
 import { CreateOrderDTO } from '../dto/create-order.dto';
 import { OrderDTO } from '../dto/order.dto';
-import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import { ApiBearerAuth, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { StocksService } from 'src/products/service/stocks.service';
 import { OrderDetailMapper } from '../mapper/order-detail.mapper';
+import { Roles } from 'src/auth/decorators/roles.decorator';
+import { JwtGuard } from 'src/auth/guards/jwt-auth.guard';
+import { RolesGuard } from 'src/auth/guards/roles.guard';
+import { Role } from 'src/customers/domain/role.enum';
 
+@Roles(Role.ADMIN, Role.CUSTOMER)
+@UseGuards(JwtGuard, RolesGuard)
```

Comment:
Good! I like how you set the both acceptable roles via the decorator and then used a chain of guards!